client/src/components/generic/Event/EventUtils.ts

```diff
@@ -0,0 +1,211 @@
+import { DateUtils } from "@/components/utils/DateUtils"
+import { Event } from "@/models/Events"
+import { MS_IN_SECOND } from "@/utils/Constants"
+import { IEventsCardPreview } from "./EventPreview/EventPreview"
+
+/**
+ * Static methods to format strings related to events
+ */
+export const EventMessages = {
+  /**
+   * Message to be displayed if sign ups have opened in relation to the sign up date
+   *
+   * @param signUpOpenDate the date object corresponding to the timestamp when the sign ups open
+   * @returns a formatted, user-readable string notifying the user of an event's sign up status
+   */
+  signUpOpen: (signUpOpenDate: Date) => {
+    const currentDate = new Date()
+    // Sign Ups have opened
+    if (signUpOpenDate <= currentDate) {
+      return `Sign ups have opened! (${EventDateFormatting.getRelativeTimeString(signUpOpenDate)})` as const
+    } else {
+      return `Sign ups open ${EventDateFormatting.getRelativeTimeString(signUpOpenDate)} (${EventDateFormatting.shortDate(signUpOpenDate)})` as const
+    }
+  },
```

Comment:
Super nice to have this although think we can have a date range or something that specifies the ending date of signups?

Reply:
the client never close sign ups anyway, so for us we just close it after the event is over

Reply:
but yea if they decide to put it in may be worth